--- ../TempleOS/TempleOSCD/Adam/ADefine.HC	2017-01-28 13:54:16.757439439 +0100
+++ Patched/Adam/ADefine.HC	2017-01-28 14:20:27.735495130 +0100
@@ -4,7 +4,7 @@
 {
   CBinFile *bfh=mem_boot_base-sizeof(CBinFile);
 
-  DefinePrint("DD_OS_NAME_VERSION","TempleOS V%0.2f",sys_os_version);
+  DefinePrint("DD_OS_NAME_VERSION","Shrine (%0.2f)",sys_os_version);
   DefinePrint("DD_TEMPLEOS_AGE","%0.1f",
 	(Now-Str2Date("8/1/2003"))/ToF64(1<<32)/CDATE_YEAR_DAYS);
 
