diff --git a/../TempleOS/TempleOSCD/Adam/ADefine.HC b/Patched/Adam/ADefine.HC
index a861bb4..c214bda 100644
--- a/../TempleOS/TempleOSCD/Adam/ADefine.HC
+++ b/Patched/Adam/ADefine.HC
@@ -4,7 +4,7 @@ U0 LoadDocDefines()
 {
   CBinFile *bfh=mem_boot_base-sizeof(CBinFile);
 
-  DefinePrint("DD_OS_NAME_VERSION","TempleOS V%0.2f",sys_os_version);
+  DefinePrint("DD_OS_NAME_VERSION","Shrine %0.2f",sys_os_version);
   DefinePrint("DD_TEMPLEOS_AGE","%0.1f",
 	(Now-Str2Date("8/1/2003"))/ToF64(1<<32)/CDATE_YEAR_DAYS);
 
