diff --git a/../TempleOS/TempleOSCD/Doc/Comm.HC b/Patched/Doc/Comm.HC
index e37e045..d6626f7 100644
--- a/../TempleOS/TempleOSCD/Doc/Comm.HC
+++ b/Patched/Doc/Comm.HC
@@ -100,6 +100,7 @@ public CComm *CommInit8n1(I64 port,I64 baud)
   return c;
 }
 
+/*
 public U0 CommPutChar(I64 port,U8 b)
 {
   CComm *c=&comm_ports[port];
@@ -110,6 +111,17 @@ public U0 CommPutChar(I64 port,U8 b)
   POPFD
   Sleep(10); //!!! Remove this line!!!	Linux echo_socket is too slow.
 }
+*/
+
+public U0 CommPutChar(I64 port,U8 b)
+{
+  I64 base=comm_ports[port].base;
+  while (!(InU8(base+UART_LSR) & 0x20))
+    Yield;
+  OutU8(base+UART_THR,b);
+  while (!(InU8(base+UART_LSR) & 0x20))
+    Yield;
+}
 
 U0 CommPutS(I64 port,U8 *st)
 {
