--- ../TempleOS/TempleOSCD/Misc/DoDistro.HC	2017-01-29 20:20:20.351074254 +0100
+++ Patched/Misc/DoDistro.HC	2017-01-29 20:10:36.681461348 +0100
@@ -31,7 +31,7 @@
   CopyTree("/Misc",	"/Distro/Misc");
 
   //To save space, optionally delete dictionary.
-  //Del("/Distro/Adam/AutoComplete/ACDefs.DATA");
+  Del("/Distro/Adam/AutoComplete/ACDefs.DATA");
   CopyTree("/Linux","/Distro/Linux");	  //You can leave this out.
   DirMk("/Distro/Tmp");
   DirMk("/Distro/Tmp/ScrnShots");
@@ -47,6 +47,6 @@
   Free(out_iso_filename);
 }
 
-MakeMyISO("/Tmp/MyDistro.ISO");
+MakeMyISO("/Tmp/ShrineDist.ISO");
 
 // Study my account examples $LK,"Cfg Strs",A="FL:::/Demo/AcctExample/TOS/TOSCfg.HC,1"$, $LK,"Update Funs",A="FL:::/Demo/AcctExample/TOS/TOSDistro.HC,1"$.
