diff --git a/../TempleOS/TempleOSCD/Once.HC b/Patched/Once.HC
index d8b4d0d..6820c80 100644
--- a/../TempleOS/TempleOSCD/Once.HC
+++ b/Patched/Once.HC
@@ -1,3 +1,7 @@
+#include "::/Apps/Lsh";
+#include "::/Apps/Pkg";
+#include "::/Misc/PalUbuntu.HC"
+
 //Place this file in /Home and change
 //anything you want.
 
@@ -52,4 +56,6 @@ U0 Tmp()
   }
 }
 
+#exe { if (SNAILNET_NATIVE_DRIVER != NULL) StreamPrint("Netcfg;\n"); }
 Tmp;
+Lsh;
