diff --git a/../TempleOS/TempleOSCD/Kernel/BlkDev/DskDrv.HC b/Patched/Kernel/BlkDev/DskDrv.HC
index f5b8ed7..7612d72 100644
--- a/../TempleOS/TempleOSCD/Kernel/BlkDev/DskDrv.HC
+++ b/Patched/Kernel/BlkDev/DskDrv.HC
@@ -175,7 +175,7 @@ I64 Let2BlkDevType(U8 drv_let)
   if ('A'<=drv_let<='B')
     return BDT_RAM;
   if ('C'<=drv_let<='L')
-    return BDT_ATA;
+    return BDT_HOSTFS;
   if ('M'<=drv_let<='P')
     return BDT_ISO_FILE_READ;
   if ('Q'<=drv_let<='S')
@@ -278,6 +278,7 @@ Bool Drv(U8 drv_let)
   switch (dv->fs_type) {
     case FSt_REDSEA:
     case FSt_FAT32:
+    case FSt_HOSTFS:
       return TRUE;
     default:
       PrintErr("File System Not Supported\n");
@@ -315,7 +316,7 @@ U8 *DrvModelNum(U8 drv_let=0)
   return res;
 }
 
-U8 blkdev_text_attr[BDT_TYPES_NUM]={BLACK,LTCYAN,WHITE,LTGREEN,LTRED,LTBLUE};
+U8 blkdev_text_attr[BDT_TYPES_NUM]={BLACK,LTCYAN,WHITE,LTGREEN,LTRED,LTBLUE,WHITE};
 U8 drv_text_attr[3]={BLACK,BLUE,RED};
 
 U8 DrvTextAttrGet(U8 drv_let=0)
