diff --git a/../TempleOS/TempleOSCD/Kernel/BlkDev/DskFile.HC b/Patched/Kernel/BlkDev/DskFile.HC
index 2eba675..d53ae9b 100644
--- a/../TempleOS/TempleOSCD/Kernel/BlkDev/DskFile.HC
+++ b/Patched/Kernel/BlkDev/DskFile.HC
@@ -28,6 +28,9 @@ U8 *FileRead(U8 *filename,I64 *_size=NULL,I64 *_attr=NULL)
 	  case FSt_FAT32:
 	    res=FAT32FileRead(dirc->dv,Fs->cur_dir,dirc->mask,&size,&attr);
 	    break;
+	  case FSt_HOSTFS:
+	    res=HostFsFileRead(dirc->dv,Fs->cur_dir,dirc->mask,&size,&attr);
+	    break;
 	  default:
 	    PrintErr("File System Not Supported\n");
 	}
