diff --git a/../TempleOS/TempleOSCD/Kernel/KUtils.HC b/Patched/Kernel/KUtils.HC
index f864990..4fef1c1 100644
--- a/../TempleOS/TempleOSCD/Kernel/KUtils.HC
+++ b/Patched/Kernel/KUtils.HC
@@ -442,16 +442,13 @@ SET_GS_BASE::
 	POP	RAX
 	RET
 SET_FS_BASE::
+	PUSH RAX
+	PUSH	0
+	PUSH	0
 	PUSH	RAX
-	PUSH	RCX
-	PUSH	RDX
-	MOV	RDX,RAX
-	SHR	RDX,32
-	MOV	ECX,IA32_FS_BASE
-	WRMSR
-	POP	RDX
-	POP	RCX
-	POP	RAX
+	PUSH	2
+	CALL	_VSYSCALL
+	POP RAX
 _RET::
 	RET
 _SET_MSR::
